File: levelOrder.py
from pyNode import Node


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


# 层序遍历
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        out = []

        if root == None:
            return []
        src = [root]

        while len(src) != 0:
            dst = []
            level = []

            while len(src) != 0:
                node = src.pop(0)

                if node.left != None:
                    dst.append(node.left)

                if node.right != None:
                    dst.append(node.right)
                level.append(node.val)
            out.append(level)
            src = dst

        return out


# N叉树的层序遍历
class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        out = []

        if root is None:
            return out
        queue = [root]

        while len(queue) > 0:
            out.append([])
            new_queue = []

            for node in queue:
                out[-1].append(node.val)

                if node.children is None:
                    continue

                for nn in node.children:
                    new_queue.append(nn)
            queue = new_queue

        return out

File: levelOrder.java
import java.util.HashSet;
import java.util.Set;
import java.util.Arrays;
import java.util.Stack;
import java.util.Queue;
import java.util.LinkedList;
import java.util.List;
import java.util.ArrayList;
import java.util.Deque;

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> out=new ArrayList<>();
        if (root == null) {
            return out;
        }
        List<Node> lis=new ArrayList<>();
        lis.add(root);
        while(lis.size()>0) {
            List<Node> newList=new ArrayList<>();
            List<Integer> tmp=new ArrayList<>();
            for(Node node:lis) {
                tmp.add(node.val);
                if(node.children!=null) {
                    for (Node nn:node.children) {
                        newList.add(nn);
                    }
                }
            }
            lis=newList;
            out.add(tmp);
        }
        return out;
    }

    public static void main(String[] args) {
        Node root=new Node(1);
        Solution sl=new Solution();
        System.out.println(sl.levelOrder(root));
    }
}

